apps/opencs/model/world/commands.cpp

```diff
@@ -235,6 +237,35 @@ void CSMWorld::CloneCommand::undo()
     mModel.removeRow (mModel.getModelIndex (mId, 0).row());
 }
 
+CSMWorld::CreatePathgridCommand::CreatePathgridCommand(IdTable& model, const std::string& id, QUndoCommand *parent)
+    : CreateCommand(model, id, parent)
+{
+    setType(UniversalId::Type_Pathgrid);
+}
+
+void CSMWorld::CreatePathgridCommand::redo()
+{
+    CreateCommand::redo();
+
+    Record<Pathgrid> record = static_cast<const Record<Pathgrid>& >(mModel.getRecord(mId));
+    record.get().blank();
+    record.get().mCell = mId;
+
+    if (!mId.empty() && mId[0]=='#')
+    {
```

Comment:
Not really. I came up with this class relatively late. There are still plenty of examples of the manual ID decoding in the code. You probably took one of these a template for your code.